### PR TITLE
Reset Password Fixes

### PR DESCRIPTION
**Pull request by nickretallack**
Like https://github.com/mattupstate/flask-security/pull/382 but with tests.

### CODE REVIEW

File: flask_security/utils.py [hunk 1]
```diff
@@ -367,7 +367,11 @@ def get_token_status(token, serializer, max_age=None):
         user = _datastore.find_user(id=data[0])
 
     expired = expired and (user is not None)
-    return expired, invalid, user
+
+    if return_data:
+        return expired, invalid, user, data
```
Comment (nickretallack) [inline flask_security/utils.py]: I hope this isn't too weird, but I did it in the name of backward compatibility and attempting to make the most minimal change possible.

**Comment by nickretallack** [comment]
Can this be accepted yet?

**Comment by mattupstate** [comment]
Thanks for the tests!

**Comment by renejahn** [comment]
any plans on releasing this in a new version anytime soon?

**Pull request merged**